include/netlink/route/link/bonding.h

```diff
@@ -26,10 +26,22 @@ extern int	rtnl_link_bond_release_ifindex(struct nl_sock *, int);
 extern int	rtnl_link_bond_release(struct nl_sock *, struct rtnl_link *);
 
 extern void	rtnl_link_bond_set_mode(struct rtnl_link *link, uint8_t mode);
+extern uint8_t	rtnl_link_bond_get_mode(struct rtnl_link *link);
+
 extern void	rtnl_link_bond_set_activeslave(struct rtnl_link *link, int active_slave);
+extern int	rtnl_link_bond_get_activeslave(struct rtnl_link *link);
+
 extern void	rtnl_link_bond_set_hashing_type (struct rtnl_link *link, uint8_t type);
+extern uint8_t	rtnl_link_bond_get_hashing_type (struct rtnl_link *link);
```

Comment:
Maybe also fix the style here
```suggestion
extern int 	rtnl_link_bond_get_hashing_type(struct rtnl_link *link, uint8_t *type);
```

Reply:
I also think so. Only returning `uint8_t` leaves no space to signal an error (or a missing attribute `-NLE_NOATTR`).

If the value itself is a `uint8_t`, then we could still do
```C
int rtnl_link_bond_get_hashing_type (struct rtnl_link *link);
```
where negative values are error codes and non-negative values are the `uint8_t` value.

For `uint32_t` types, that doesn't work. So it needs
```C
extern int rtnl_link_bond_get_miimon(struct rtnl_link *link, uint32_t *miimon);
```
to separately signal the error/NOATTR and the value.


I think `int rtnl_link_bond_get_hashing_type (struct rtnl_link *link);` is still less convenient to use, because you need
```C
    r = rtnl_link_bond_get_hashing_type(link);
    if (r >= 0)
         hashing_type = r;
    else {
          /* handle error */
    }
```
instead of
```C
     r = rtnl_link_bond_get_hashing_type(link, &hashing_type);
     if (r < 0) {
         /* handle error */
     }
```

So just have getters return an `int` (zero or negative error code) and the out value separately. Ensure that the out value is touched, if-and-only-if there is no error.